playbooks/reboot_cluster.yml

```diff
@@ -1,48 +1,124 @@
 ---
-- hosts: all:!db:!vyatta-*
+- hosts: controller
   serial: 1
   tasks:
   - name: Reboot
     command: shutdown -r now
     async: 0
     poll: 0
-    ignore_errors: true
+    failed_when: false
 
   - name: Wait for SSH to come back
     wait_for: host={{ ansible_default_ipv4.address }}
               port={{ ansible_ssh_port|default(22) }}
               delay=15
     delegate_to: "{{ groups['controller']|first }}"
 
-- hosts: db_arbiter
+  - name: cluster health check
+    command: /etc/sensu/plugins/percona-cluster-size.rb -d /root/.my.cnf --expected 3 --criticality critical
+    register: cstat
+    until: cstat | succeeded
+    retries: 5
+    run_once: true
+
+- hosts: compute:!controller
   serial: 1
   tasks:
-  - name: Ensure Garbd is up before rebooting database servers
+  - name: Reboot
+    command: shutdown -r now
+    async: 0
+    poll: 0
+    failed_when: false
+
+  - name: Wait for SSH to come back
     wait_for: host={{ ansible_default_ipv4.address }}
-              port=4567
+              port={{ ansible_ssh_port|default(22) }}
               delay=15
-    delegate_to: "{{ groups['controller']|first }}"
 
-- hosts: db
+  - name: check nova services
+    command: /etc/sensu/plugins/check-nova-services.sh
+    register: result
+    until: result | succeeded
+    retries: 10
+    delegate_to: "{{ item }}"
+    with_items: "{{ groups['controller']|first }}"
```

Comment:
I think you can skip the `with_items` here and just do a straight `delegate_to`.